semigroups.cabal

```diff
@@ -128,6 +136,9 @@ library
     if flag(bytestring)
       build-depends: bytestring >= 0.9 && < 1
 
+    if flag(bytestring-builder) && impl(ghc < 7.8)
```

Comment:
BTW @hvr, if `ghc` can't be used as a proxy for package versions anymore, then what *is* the proper way to predicate a `.cabal` file based on what version of a `package` is available? I ask since there is [currently](https://github.com/ekmett/semigroups/blob/ef1a256e14bce5ce3bd387fbf871ddbb874360e2/semigroups.cabal#L111) an `if impl(ghc < 7.11.20151002)` block used to conditionally determine whether to export the `Data.Semigroup` module, but by your own admission, we should be checking whether `base-4.9.0.0` is being used, not whether GHC 8 is being used.